### PR TITLE
[NVIDIA XLA:GPU] Fused MHA Support in XLA GPU: SE

### PR DESCRIPTION
**Pull request by AyanmoI**
cuDNN currently supports the following patterns for Multi-headed attention:

BMM1 - BMM2
BMM1 - Scale - Bias - Mask - Softmax - BMM2
BMM1 - Scale - Bias - Mask - Softmax - Dropout - BMM2
BMM1 - Scale - Mask - Softmax - BMM2
BMM1 - Scale - Mask - Softmax - Dropout - BMM2
BMM1 - Softmax - Dropout - BMM2

This PR adds support for the stream executor for these patterns.

### CODE REVIEW

**Comment by AyanmoI** [comment]
@akuegel @cheshire Based on https://github.com/tensorflow/tensorflow/pull/60139#issuecomment-1486799319, I created this new PR.

File: tensorflow/compiler/xla/stream_executor/dnn.h [hunk 1]
```diff
@@ -152,6 +155,75 @@ enum class RnnDirectionMode {
 // DepthToSpace comment for more information.
 enum class DepthToSpaceLayout { DepthHeightWidth };
 
+class TensorDescriptor {
```
Comment (cheshire) [inline tensorflow/compiler/xla/stream_executor/dnn.h:158]: This is an almost exact clone of xla::Shape, right? I wonder if it's possible to unify those better, I guess lack of SE->XLA dependencies prevents it, but then do we need a whole new abstraction there?

Comment (AyanmoI) [reply tensorflow/compiler/xla/stream_executor/dnn.h:158]: > This is an almost exact clone of xla::Shape, right? I wonder if it's possible to unify those better, I guess lack of SE->XLA dependencies prevents it, but then do we need a whole new abstraction there?

You are right..the information here can be extracted form xla::Shape but as you said there is no way to have that dependency and reuse xal::Shape here. However, this is much more light weight than `Shape` so it's not too bad. I looked to see if there is anything that already exists and fulfills this requirement but couldn't find any that can be reused.

**Review by cheshire — APPROVED** [review]

**Comment by AyanmoI** [comment]
@akuegel Possible to merge this now? Hopefully no more issues :)
